config/buffer_period.go

```diff
@@ -86,14 +77,39 @@ func (c *BufferPeriodConfig) Merge(o *BufferPeriodConfig) *BufferPeriodConfig {
 	return r
 }
 
-// Finalize ensures there no nil pointers.
-func (c *BufferPeriodConfig) Finalize() {
-	if c.Enabled == nil {
-		c.Enabled = Bool(TimeDurationPresent(c.Min))
+// Finalize ensures there no nil pointers. For nil pointers, use 'parent'
+// BufferPeriodConfig and other default values as necessary.
+//
+// Finalizing global-level buffer period, use DefaultBufferPeriodConfig() as
+// parent. Finalizing task-level buffer period, use global-level buffer period
+// as parent.
+func (c *BufferPeriodConfig) Finalize(parent *BufferPeriodConfig) {
```

Comment:
For the above comment regarding inputs to Finalize, I wonder if this should be included here? I get that we're trying to instruct how this method should be used, but it seems like a comment that might be better suited for where this method is used (like config/config.go)? 

To me it seems like Finalize itself is more agnostic towards global-level vs task-level it just uses the parent as a means for setting default values when nil is present.